src/js/__tests__/auth/forOwnerOnly.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import '@testing-library/jest-dom/extend-expect'
+
+import { render } from '@testing-library/react'
+import forOwnerOnly from '../../auth/forOwnerOnly'
+import usePermissions from '../../hooks/auth/usePermissions'
+import { IUserProfile, WithOwnerProfile } from '../../types/User'
+
+// Mock the usePermissions hook
+jest.mock('../../hooks/auth/usePermissions')
+
+const dummyOwnerProfile: IUserProfile = {
+  uuid: '12345678-1234-1234-1234-123456789012',
+  roles: ['user'],
+  loginsCount: 1,
+  name: 'John Doe',
+  nick: 'johndoe',
+  bio: 'Climber and outdoor enthusiast',
+  website: 'https://johndoe.com',
+  ticksImported: true,
+  collections: {
+    climbCollections: {
+      'John\'s Climbs': ['climb1', 'climb2', 'climb3']
+    },
+    areaCollections: {
+      'John\'s Areas': ['area1', 'area2', 'area3']
+    }
+  },
+  email: 'john@example.com',
+  avatar: 'https://example.com/john/avatar.jpg',
+  authProviderId: 'auth0|123456789012345678901234'
+}
+// Create a mock component to be used with forOwnerOnly
+const MockComponent: React.FC<WithOwnerProfile> = () => <div>Mock Component</div>
+
+// Create a component wrapped with forOwnerOnly
+const WrappedComponent = forOwnerOnly(MockComponent)
+
+describe('forOwnerOnly', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should render the component if isAuthorized is true', async () => {
+    (usePermissions as jest.Mock).mockReturnValue({ isAuthorized: true })
```

Comment:
// jest.Mock(usePermission) 
not sure the actual constructor here so pseudocode above. the pseudocode would in theory should have the correct prototype, previously you are just adding the type which does nothing. The only reason it is working is because of some jest global environment magic.